tests/test_callbacks.py

```diff
@@ -0,0 +1,153 @@
+"""Test for callbacks
+"""
+
+from transformers import (
+    Seq2SeqTrainer,
+    Seq2SeqTrainingArguments,
+    T5Tokenizer,
+    T5ForConditionalGeneration,
+    DataCollatorForSeq2Seq,
+)
+
+from datasets import load_dataset
+
+from deepeval.callbacks.huggingface import DeepEvalCallback
+from deepeval.metrics import HallucinationMetric, AnswerRelevancyMetric
+from deepeval.dataset import EvaluationDataset, Golden
+
+import os
+
+os.environ["TRANSFORMERS_NO_ADVISORY_WARNINGS"] = "true"
+os.environ["OPENAI_API_KEY"] = "API-KEY"
+
+
+def create_prompt(row):
+    """Merge Context and Question into a single string"""
+    contexts = row["context"]["contexts"]
+    question = row["question"]
+    prompt = f"""{'CONTEXT: ' + str("; ".join(contexts)) if contexts else ''}
+            QUESTION: {question}
+            ANSWER:"""
+    return {"input": prompt, "response": row["long_answer"]}
+
+
+def prepare_dataset(tokenizer, tokenizer_args):
+    dataset = load_dataset("pubmed_qa", "pqa_labeled")
+    merged_dataset = dataset.map(
+        create_prompt,
+        remove_columns=[
+            "question",
+            "context",
+            "long_answer",
+            "pubid",
+            "final_decision",
+        ],
+    )
+
+    def tokenize_text(dataset, padding="max_length"):
+        model_input = tokenizer(dataset["input"], **tokenizer_args)
+        response = tokenizer(dataset["response"], **tokenizer_args)
+
+        if padding == "max_length":
+            response["input_ids"] = [
+                [(l if l != tokenizer.pad_token_id else -100) for l in label]
+                for label in response["input_ids"]
+            ]
+
+        model_input["labels"] = response["input_ids"]
+        return model_input
+
+    tokenized_dataset = merged_dataset.map(
+        tokenize_text, remove_columns=["input", "response"]
+    )
+    tokenized_dataset = tokenized_dataset.map(
+        lambda x: {
+            "input_ids": x["input_ids"][0],
+            "labels": x["labels"][0],
+            "attention_mask": x["attention_mask"][0],
+        }
+    )
+    return dataset, merged_dataset, tokenized_dataset
+
+
+def create_deepeval_dataset(dataset, sample_size):
+    eval_dataset = [dataset[row] for row in range(5, 10)]
```

Comment:
Why the range (did you mean to use sample size instead?)

Reply:
`dataset` that I am passing here is the fine-tuning dataset itself. And the `sample_size` was meant to randomly pick that many data points from `dataset` for deepeval evaluation. That part lags, Will fix it today.

Reply:
I see, thanks!

Reply:
Fixed this one. Please check it out!